src/arrayMethodJoin.js

```diff
@@ -4,8 +4,32 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedArray = '';
+
+    if (this[0] !== null) {
+      joinedArray += this[0];
+    }
```

Comment:
is this check necessary?

Reply:
When I remove this check, two tests are failed:

1)  test(`Should work when array has null as a first element`, () => {
      78 |   expect([null, null, 1, undefined, 0, NaN, ''].join2(','))
    > 79 |     .toBe(',,1,,0,NaN,');
         |      ^
         
2) test(`Should work when array contains a string 'null'`, () => {
      83 |   expect([null, null, 1, undefined, 0, 'null', NaN, ''].join2(','))

Reply:
> My point is that with this change you probably won't need to have this[0] in the array initially, which will make the code cleaner

related to this